src/Bundle/DependencyInjection/Driver/DriverProvider.php

```diff
@@ -13,41 +13,39 @@
 
 namespace Sylius\Bundle\ResourceBundle\DependencyInjection\Driver;
 
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineODMDriver;
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrineORMDriver;
-use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Doctrine\DoctrinePHPCRDriver;
 use Sylius\Bundle\ResourceBundle\DependencyInjection\Driver\Exception\UnknownDriverException;
-use Sylius\Bundle\ResourceBundle\SyliusResourceBundle;
 use Sylius\Component\Resource\Metadata\MetadataInterface;
 use Webmozart\Assert\Assert;
 
 final class DriverProvider
 {
-    /** @var DriverInterface[] */
+    /** @var array<string, array<string, string>> */
     private static array $drivers = [];
 
+    public static function build(array $drivers): void
+    {
+        self::$drivers = $drivers;
+    }
+
     /**
      * @throws UnknownDriverException
      */
     public static function get(MetadataInterface $metadata): DriverInterface
     {
         $type = $metadata->getDriver();
 
-        if (isset(self::$drivers[$type])) {
-            return self::$drivers[$type];
-        }
+        /** @var class-string|null $class */
+        $class = self::$drivers[$type]['class'] ?? null;
```

Comment:
This line is somewhat misleading as `$class` is made nullable, while only `$drivers[$type]` is nullable and `$drivers[$type]['class']` should be asserted

```php
if (isset(self::$drivers[$type])) {
    Assert::notNull(self::$drivers[$type]['class'], sprintf('Class must be present for driver "%s" configuration', $type));
    $class = self::$drivers[$type]['class'];
    
    return new $class();
```

Reply:
IMHO this kind of assertion must be done into a `DriverMetadata` class like I suggest earlier.